pkg/controller/proxyconfig/validation.go

```diff
@@ -52,5 +60,85 @@ func (r *ReconcileProxyConfig) ValidateProxyConfig(proxyConfig *configv1.ProxySp
 		}
 	}
 
+	if isSpecReadinessEndpointsSet(proxyConfig) {
+		for _, endpoint := range proxyConfig.ReadinessEndpoints {
+			scheme, err := validation.URI(endpoint)
+			if err != nil {
+				return fmt.Errorf("invalid URI for endpoint '%s': %v", endpoint, err)
+			}
+			switch {
+			case scheme == proxyHTTPScheme:
+				// TODO: Add case for proxyHTTPSScheme when CA support is merged.
+				if err := validateHTTPReadinessEndpoint(proxyConfig.HTTPProxy, endpoint); err != nil {
+					return fmt.Errorf("readinessEndpoint probe failed for endpoint '%s'", endpoint)
+				}
+			default:
+				// TODO: Update error to include proxyHTTPSScheme when CA support is merged.
+				return fmt.Errorf("readiness endpoints requires a '%s' URI sheme", proxyHTTPScheme)
+			}
+		}
+	}
+
+	return nil
+}
+
+// validateHTTPReadinessEndpoint validates an http readinessEndpoint endpoint.
+func validateHTTPReadinessEndpoint(httpProxy, endpoint string) error {
+	if err := validateHTTPReadinessEndpointWithRetries(httpProxy, endpoint, proxyProbeMaxRetries); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+// validateHTTPReadinessEndpointWithRetries tries to validate an http
+// endpoint in a finite loop and returns the last result if it never succeeds.
+func validateHTTPReadinessEndpointWithRetries(httpProxy, endpoint string, retries int) error {
+	var err error
+	for i := 0; i < retries; i++ {
+		if i != 0 {
+			time.Sleep(proxyProbeWaitTime)
+		}
```

Comment:
you could just move this after the "if err=nil return" check and not have to look at `i`.

Reply:
ack